cosmos_rl/policy/trainer/sft_trainer.py

```diff
@@ -849,8 +849,19 @@ def train(self):
                 # save checkpoint
                 if (
                     self.config.train.ckpt.enable_checkpoint
-                    and self.train_step % self.config.train.ckpt.save_freq == 0
-                    and self.train_step > 0
+                    and (
+                        (
+                            self.train_step % self.config.train.ckpt.save_freq == 0
+                            and self.train_step > 0
+                        )
+                        or (
+                            self.config.train.ckpt.save_freq_in_epoch > 0
+                            and self.train_step % len(self.train_data_loader) == 0
```

Comment:
can we do sth. like below in the `__init__` ? (or somewhere during initialization / before training start)

```python
if self.config.train.ckpt.save_freq_in_epoch is not None:
    self._save_freq = len(self.train_data_loader)
elif self.config.train.ckpt.save_freq is not None:
    self._save_freq = self.config.train.ckpt.save_freq
else:
    # only save at the end of training
    self._save_freq = len(self.train_data_loader) * self.epoch
```

Reply:
Done with these modifications.